openedx/core/djangoapps/user_authn/views/utils.py

```diff
@@ -108,3 +113,60 @@ def get_mfe_context(request, redirect_to, tpa_hint=None):
         'countryCode': country_code,
     })
     return context
+
+
+def generate_username(username_initials):
+    """
+    Generate a username based on initials and current date.
+
+    Args:
+    - username_initials (str): Initials used for constructing the username.
+
+    Returns:
+    - str: Generated username.
+    """
+    current_year = datetime.now().year % 100
+    current_month = datetime.now().month
+
+    random_string = ''.join(random.choices(
+                            string.ascii_letters +
+                            string.digits,
+                            k=settings.RANDOM_USERNAME_STRING_LENGTH))
+
+    username = f"{username_initials}_{current_year:02d}{current_month:02d}_{random_string}"
+    return username
+
+
+def generate_username_from_request_payload(data):
+    """
+    Generate a username based on the provided data.
+
+    Args:
+    - data (dict): Dictionary containing user data, including 'first_name', 'last_name', or 'name'.
+
+    Returns:
+    - str: Generated username.
+    """
+    try:
+        if 'first_name' in data and 'last_name' in data:
```

Comment:
We need to verify if 'first_name' and 'last_name' are included in the `REGISTRATION_EXTRA_FIELDS` list. This is because SSO workflows almost always provide these fields. However, if the registration form doesn't have these fields enabled and has the full name field, then that full name should be used to generate name initials.

Reply:
I have tested this scenario, and in the payload, neither the first_name nor the last_name is included. Furthermore, the username has been removed from the front-end application. Therefore, this case will not occur.